lib/CustomFields/pages/EditCustomFieldsRecord/EditCustomFieldsRecord.js

```diff
@@ -0,0 +1,148 @@
+import React from 'react';
+import { PropTypes } from 'prop-types';
+import { connect } from 'react-redux';
+import { FormattedMessage } from 'react-intl';
+import { chunk } from 'lodash';
+import {
+  Accordion,
+  Callout,
+  Col,
+  Row,
+  InfoPopover,
+} from '@folio/stripes-components';
+
+import {
+  fieldComponents,
+  rowShapes,
+} from '../../constants';
+import {
+  selectModuleId,
+  selectOkapiData,
+} from '../../selectors';
+import {
+  useCustomFieldsFetch,
+  useSectionTitleFetch,
+  useLoadingErrorCallout,
+  valuesValidation,
+} from '../../utils';
+
+const propTypes = {
+  accordionId: PropTypes.string.isRequired,
+  backendModuleId: PropTypes.string,
+  backendModuleName: PropTypes.string.isRequired,
+  columnCount: PropTypes.number,
+  entityType: PropTypes.string.isRequired,
+  expanded: PropTypes.bool.isRequired,
+  fieldComponent: PropTypes.node.isRequired,
+  okapi: PropTypes.shape({
+    tenant: PropTypes.string.isRequired,
+    token: PropTypes.string.isRequired,
+    url: PropTypes.string.isRequired,
+  }).isRequired,
+  onToggle: PropTypes.func.isRequired,
+};
+
+const defaultProps = {
+  columnCount: 4,
+};
+
+const EditCustomFieldsRecord = ({
+  accordionId,
+  onToggle,
+  expanded,
+  okapi,
+  backendModuleId,
+  backendModuleName,
+  entityType,
+  fieldComponent,
+  columnCount,
+}) => {
+  const {
+    customFields,
+    customFieldsLoaded,
+    customFieldsFetchFailed,
+  } = useCustomFieldsFetch(okapi, backendModuleId, entityType);
+  const {
+    sectionTitle,
+    sectionTitleLoaded,
+    sectionTitleFetchFailed,
+  } = useSectionTitleFetch(okapi, backendModuleName.toUpperCase());
+  const { calloutRef } = useLoadingErrorCallout(customFieldsFetchFailed || sectionTitleFetchFailed);
+  const dataIsLoaded = sectionTitleLoaded && customFieldsLoaded;
+  const customFieldsIsVisible = customFields?.length > 0 && customFields?.some(customField => customField.visible);
+  const accordionIsVisible = dataIsLoaded && customFieldsIsVisible;
+
+  const Field = fieldComponent;
+  const formatedCustomFields = chunk(customFields, columnCount);
+  const columnWidth = rowShapes / columnCount;
+  const defaultLabel = <FormattedMessage id="stripes-smart-components.customFields.recordAccordion.defaultName" />;
+
+  return (
+    <>
+      {accordionIsVisible
+        ? (
+          <Accordion
+            open={expanded}
+            id={accordionId}
+            onToggle={onToggle}
+            label={sectionTitle.value || defaultLabel}
```

Comment:
I don't think we need to have a default label. The accordion won't be rendered if no section title is set